python/pyarrow/_orc.pyx

```diff
@@ -388,8 +388,11 @@ cdef class ORCWriter(_Weakrefable):
         object sink
         unique_ptr[ORCFileWriter] writer
         shared_ptr[COutputStream] rd_handle
```

Comment:
Can you take the opportunity to rename this? The name is obviously wrong.

Reply:
For reference, in the cython ParquetWriter, we have

```
        shared_ptr[COutputStream] sink
        bint own_sink
```

and this `own_sink` is defined based on the input sink (file path we open ourselves vs already open file-like object-, and the use that to determine whether `close()` should close it or not.

We can maybe follow similar pattern here?